### ISSUE TITLE
2024.07  Release Tracking

### ISSUE THREAD

**Issue by rtsisyk** [organicmaps/organicmaps#8407]
Expected Store Submission: Sun 07 Jul (around 20:00 GMT)
Expected Full Rollout to Users: Fri 12, Jul
Release Manager: @rtsisyk

- [x] Alpha reminders
 	- [x] PRs for alpha reminder | TBD  | Wed, Jun 19
 	- [x] The release notes reminder| TBD | Wed, Jun 19
 	- [x]  PRs for alpha reminder | TBD | Fri, Jun 21
 	- [x] The release notes reminder| TBD | Fri, Jun 21
- [x] Alpha release | TBD | Sat, Jun 22 - Sun, Jun 23
 	- [x] [No release blockers](https://github.com/organicmaps/organicmaps/milestone/9)  | TBD
 	- [x] Alpha release notes are ready in Etherpad | TBD
 	- [x] Android smoke-test is OK (15 minutes) | TBD
 	- [x] iOS smoke-test is OK  (15 minutes) | TBD
  	- [x] A new alpha branch is created from the master
 	     - [x] GitHub Branch Protection is configured for the new branch
 	     - [x] GitHub Environments are configured  for the new branch
  	- [x] Android Alpha is published on Firebase AppTester for testing
 	- [x] Android Alpha is published on Telegram for testing
 	- [x] iOS Alpha is submitted to Apple TestFlight for review
 	- [x] iOS Alpha is approved in Apple TestFlight
 	     - [ ] iOS Alpha TestFlight link is published on Telegram for testing
- [x] Release Candidate reminders
 	- [x] Data generation reminder | TBD  | **Wed, Jun 26**
 	- [x] The release notes reminder |TBD | Wed, Jun 26
 	- [x] ASO reminder | TBD  |  Wed, Jun 26
- [ ] Release Candidate | @rtsisyk | Sun, Jul 7
 	- [x]  [No release blockers in master](https://github.com/organicmaps/organicmaps/milestone/9)  | @rtsisyk
 	- [x] A PR with the actual release notes is raised | TBD | Sun, Jul 7
 	- [ ] Android smoke-test is OK (15 minutes) | @rtsisyk
 	- [ ] iOS smoke-test is OK (15 minutes) | @rtsisyk
 	- [x] A new release branch is created from the master
 	     - [x] GitHub Branch Protection is configured for the new branch
 	     - [x] GitHub Environments are configured  for the new branch
 	- [x] A PR with the actual data is raised | @vng | Fri, Jul 5
 	- [ ] RC Sign-off
 	     - [ ] Android is tested by monkey | @rtsisyk
 	     - [x] Android is tested manually by @Jean-BaptisteC
 	     - [x] Android is tested by @rtsisyk
 	     - [ ] Android is tested by @vng
 	     - [ ] iOS is tested by @kirylkaveryn 	 
 	     - [x] iOS is tested by @rtsisyk
 	     - [x] iOS is tested by @vng
 	- [x] The PR with new data is merged into the master
 	- [x] The PR with release notes is merged into the master
 	- [x] Android RC is submitted to Google Play Closed Beta for review
 	- [x] Android RC is published on GitHub as a pre-release
 	- [x] Android RC APK from Google Play is uploaded to GitHub
 	- [x] Android RC is published on Firebase AppTester for testing
 	- [ ] Android RC is published on Telegram for testing
 	- [x] F-Droid build is triggered | @pastk and @rtsisyk to figure out how to check
 	- [x] iOS RC is submitted to Apple TestFlight for review
 	- [ ] Data is synchronized to all servers
 	- [x] Metaserver is updated to include new data version
  	- [x] iOS RC TestFlight is approved | @rtsisyk | Tue, Jul 09
 	     - [ ] iOS RC TestFlight link is published on Telegram for testing
  	- [x] Android RC is approved for Google Play Closed Beta | @rtsisyk |  Tue, Jul 09
 	     - [ ] Google Play Managed Publishing Check
- [ ] Production Release | Tue, Jul 09
 	- [x] iOS RC AppStore metadata is updated | @yanatalia | Tue, Jul 09
 	- [x] iOS RC is sent for review to App Store production with automatic 7 days phased rollout
 	- [x] Android RC is sent for review to Google Play Production with automatic 10% phased rollout| @rtsisyk
 	- [x] Google Play Managed Publishing Check
 	- [x] Android RC is sent for review to Huawei AppGallery with manual release (TBD) | @rtsisyk
  	- [x] Metaserver is updated to include new app versions  | @rtsisyk | Wed, Tue 9
  	- [x] Google Play 20% phased rollout | @rtsisyk | Wed, Tue 10
  	- [x] The release final sign-off | Wed, Tue 10
 	     - [ ] Firebase reports by @rtsisyk
 	     - [ ] Google Play reports by @rtsisyk
 	     - [ ] TestFlight reports by @kirylkaveryn
 	     - [ ] Email & Telegram reports by @biodranik
  	- [x] AppStore 100% rollout | TBD | Fri, Jul 12
  	- [x] Google Play 100% rollout | TBD | Fri, Jul 12
  	- [ ] App Gallery 100% rollout | TBD | Fri, Jul 12
  	- [x] GitHub - remove "pre-release" and mark the release as "latest" | TBD | Fri, Jul  12
  	- [x] F-Droid check | TBD | Fri, Jul  12
  	- [ ] The release branch is merged into master | @rtsisyk | Fri, Jul  12
- [ ] Announcements
  	- [ ] Telegram announcement | @biodranik 
  	- [ ] Mastodon announcement | @biodranik
  	- [ ] Facebook announcement | @biodranik 
  	- [ ] Website announcement | @biodranik

**Comment by pastk** [comment]
> @organicmaps/maintainers are we OK with proposed dates?

LGTM, though I won't be available since ~1st of July.

Please move `A PR with the actual release notes is raised` to be after `A new release branch is created from the master` (as relnotes will target the branch).

**Comment by biodranik** [comment]
A better plan is to release an alpha/beta version on Sun 30, to include and properly test new features and fixes. And to make the release in a week, after fixing all issues.

@organicmaps/contributors can you please help with PRs reviewing and finishing them? Please add (almost)ready PRs [to this milestone](https://github.com/organicmaps/organicmaps/milestone/8). Another important task is to start updating release notes for the upcoming release, and a list of features to test for alpha/beta. Here are some: https://github.com/organicmaps/organicmaps/milestone/8

**Comment by pastk** [comment]
> A better plan is to release an alpha/beta version on Sun 30, to include and properly test new features and fixes. And to make the release in a week, after fixing all issues.
> 

What's the point of postponing next release?

Releases do not block development nowadays, so work on not-ready-yet features could be continued independently.
IMO its better to release according to the initial plan and deliver regular data updates.

**Comment by biodranik** [comment]
One of the reasons is that we have many important changes to test (and will have more merged this week), including the changed behavior for single/long tap.

We always used betas before, and it was very, very helpful in discovering issues before they hit production.

**Comment by pastk** [comment]
> 
> Please move `A PR with the actual release notes is raised` to be after `A new release branch is created from the master` (as relnotes will target the branch).

Also I thought we agreed on creating a release branch before starting data generation, no?
(so that no data-affecting changes get merged after the gen start)

**Comment by pastk** [comment]
What's up with the release? Who's going to push it?

@rtsisyk I understand you'll be unavailable soon?

**Comment by kirylkaveryn** [comment]
iOS tested and works as intended.
No issues found.

**Comment by pastk** [comment]
F-Droid build failed on git fetch:
```
From https://github.com/organicmaps/organicmaps

* [new branch]            ab-android-gradle-update -> origin/ab-android-gradle-update

...

error: cannot lock ref 'refs/remotes/origin/alpha/2024.07.02': 'refs/remotes/origin/alpha' exists; cannot create 'refs/remotes/origin/alpha/2024.07.02'

! [new branch]            alpha/2024.07.02      -> origin/alpha/2024.07.02  (unable to update local ref)

* [new branch]            ios/add-dontation-section-to-pp -> origin/ios/add-dontation-section-to-pp

...

error: some local refs could not be updated; try running

'git remote prune origin' to remove any old, conflicting branches
```

I had a similar issue on my local setup and I had to manually remove the old alpha branch.

**Comment by pastk** [comment]
> F-Droid build failed on git fetch https://monitor.f-droid.org/builds/log/app.organicmaps/24070803

https://gitlab.com/fdroid/fdroidserver/-/merge_requests/1494

**Comment by pastk** [comment]
F-Droid build issue had been fixed and the latest build was successful. Waiting for a sign & publish..

**Comment by pastk** [comment]
F-Droid build had been released!
